sklearn/metrics/_plot/roc_curve.py

```diff
@@ -333,20 +382,437 @@ def from_predictions(
+            )
+
+        train_size, test_size = (
+            len(cv_results["indices"]["train"][0]),
+            len(cv_results["indices"]["test"][0]),
         )
 
-        return viz.plot(ax=ax, name=name, **kwargs)
+        if _num_samples(X) != train_size + test_size:
+            raise ValueError(
+                "X does not contain the correct number of samples. "
+                f"Expected {train_size + test_size}, got {_num_samples(X)}."
+            )
+
+        # TODO: it should be remove once #23073 is merged since we delegate this check
+        # to `_get_response_values_binary`.
+        if not all(is_classifier(estimator) for estimator in cv_results["estimator"]):
+            raise ValueError(
+                "The estimators in cv_results['estimator'] must be fitted classifiers."
+            )
+
+        if fold_name is None:
+            # create an iterable of the same length as the number of ROC curves
+            fold_name_ = [None] * len(cv_results["estimator"])
+        elif fold_name is not None and len(fold_name) != len(cv_results["estimator"]):
+            raise ValueError(
+                "When `fold_name` is provided, it must have the same length as "
+                f"the number of ROC curves to be plotted. Got {len(fold_name)} names "
+                f"instead of {len(cv_results['estimator'])}."
+            )
+        else:
+            fold_name_ = fold_name
+
+        displays = []
+        for fold_id, (estimator, test_indices, name) in enumerate(
+            zip(cv_results["estimator"], cv_results["indices"]["test"], fold_name_)
+        ):
+            y_true = _safe_indexing(y, test_indices)
+            y_pred = _get_response(
+                _safe_indexing(X, test_indices),
+                estimator,
+                response_method=response_method,
+                pos_label=pos_label,
+            )[0]
+
+            displays.append(
+                cls._create_display_from_predictions(
+                    y_true=y_true,
+                    y_pred=y_pred,
+                    sample_weight=sample_weight,
+                    drop_intermediate=drop_intermediate,
+                    pos_label=pos_label,
+                    name=f"ROC fold #{fold_id}" if name is None else name,
+                )
+            )
+
+        viz = MultiRocCurveDisplay(displays=displays)
+        return viz.plot(
+            ax=ax,
+            kind=kind,
+            fold_name=fold_name,
+            fold_line_kw=fold_line_kw,
+            aggregate_name=aggregate_name,
+            aggregate_line_kw=aggregate_line_kw,
+            aggregate_uncertainty_kw=aggregate_uncertainty_kw,
+        )
+
+
+class MultiRocCurveDisplay:
+    """Multiple ROC curves visualization.
+
+    It is recommended to use
+    :meth:`~sklearn.metrics.RocCurveDisplay.from_cv_results` to create a
+    :class:`~sklearn.metrics.MultiRocCurveDisplay`. All parameters are stored as
+    attributes.
+
```

Comment:
If we don't want the users to directly use it, we can make it private instead, no ?

Reply:
We do the same mention in over display. However, I am not against making the class private. However, I am thinking that it could be a handy display to plot several ROC curves even if they don't come from cross-validation.

Reply:
For the other display we discourage calling `plot` directly. Here it's the whole class. If we find it useful on its own, maybe the message should be something like "if you want to plot multiple roc curves from cv results, you should use ... instead".

I'm guessing that if we want to add from_cv_results to other displays, we'll have to make a `Multi...Display` for each. Do you think we could at least factor some code into a Multi...Mixin ?

Reply:
> For the other display we discourage calling plot directly. Here it's the whole class. If we find it useful on its own, maybe the message should be something like "if you want to plot multiple roc curves from cv results, you should use ... instead".

I see. I can alter the code.

> I'm guessing that if we want to add from_cv_results to other displays, we'll have to make a Multi...Display for each. Do you think we could at least factor some code into a Multi...Mixin ?

Indeed. I would postpone this introduction to the next display to see how much in common we will have between displays.